Sources/CodegenCLI/Extensions/VersionChecker.swift

```diff
@@ -40,42 +40,63 @@ enum VersionChecker {
       return fileManager.base.contents(atPath: path)
     }
 
-    // When using SPM via Xcode, the `Package.resolved` file is nested inside the `.xcproject`
-    // package. Since we don't know the name of your Xcode project, we just look for the first
-    // `.xcproject` file in project's root directory that depends on Apollo. This may not be
-    // 100% fool-proof, but it should be accurate in almost all cases.
-    func findInXcodeProject(named projectFileName: String) -> Data? {
-      let projectPackagePath =
-      "\(projectFileName)/project.xcworkspace/xcshareddata/swiftpm/\(Package_resolved)"
-
+    // When using SPM via Xcode, the `Package.resolved` file is nested inside either the
+    // `.xcworkspace` or the `.xcproject` package. Since we don't know the name of your project,
+    // we just look for the first workspace or project that depends on Apollo, prioritising
+    // workspaces. This may not be 100% fool-proof, but it should be accurate in almost all cases.
+    func findInXcodeWorkspaces() -> PackageResolvedModel? {
```

Comment:
These two functions appear identical except for the file suffix and package path. I think it would be a good improvement to extract the bulk of the logic into a single function that has parameters for those two pieces that are different.